pkg/controller/controller_instance.go

```diff
@@ -1237,6 +1311,27 @@ func (c *controller) clearServiceInstanceCurrentOperation(toUpdate *v1alpha1.Ser
 	toUpdate.Status.CurrentOperation = ""
 	toUpdate.Status.OperationStartTime = nil
 	toUpdate.Status.AsyncOpInProgress = false
+	toUpdate.Status.OrphanMitigationInProgress = false
 	toUpdate.Status.LastOperation = nil
 	toUpdate.Status.ReconciledGeneration = toUpdate.Generation
 }
+
+// setServiceInstanceStartOrphanMitigation sets the fields of the instance's
+// Status to indicate that orphan mitigation is starting. The Status is *not*
+// recorded in the registry.
+func setServiceInstanceStartOrphanMitigation(toUpdate *v1alpha1.ServiceInstance) {
```

Comment:
I wonder whether we should be clearing the CurrentOperation here? My understanding of what that field represents is that the CurrentOperation is what operation we are currently trying to send to the Broker. We have stopped the current operation and are starting deprovision via orphan mitigation.

Reply:
I can see it going either way. I had thought it was more describing the longer-term operation the user had requested from the controller. @pmorie, what are your opinions?